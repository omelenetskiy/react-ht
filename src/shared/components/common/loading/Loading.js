import React, { Suspense } from "react";

const Loading = Component => props => (
  <Suspense fallback={<div>Loading...</div>}>
    <Component {...props} />
  </Suspense>
);

export default Loading;
