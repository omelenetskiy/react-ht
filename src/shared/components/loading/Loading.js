import React from 'react';
import styled from 'styled-components';

const StyledLoading = styled.div`
  height: 100vh;
  display: inline-flex;
  align-items: center;
  justify-content: center;

  .spinner {
    margin: 100px auto;
    width: 50px;
    height: 65px;
    text-align: center;
    font-size: 10px;
  }

  .spinner > div {
    background-color: #fff;
    height: 100%;
    width: 6px;
    margin: 0 3px 0 0;
    display: inline-block;

    -webkit-animation: sk-stretchdelay 1.2s infinite ease-in-out;
    animation: sk-stretchdelay 1.2s infinite ease-in-out;
  }

  .spinner .rect2 {
    -webkit-animation-delay: -1.1s;
    animation-delay: -1.1s;
  }

  .spinner .rect3 {
    -webkit-animation-delay: -1s;
    animation-delay: -1s;
  }

  .spinner .rect4 {
    -webkit-animation-delay: -0.9s;
    animation-delay: -0.9s;
  }

  .spinner .rect5 {
    -webkit-animation-delay: -0.8s;
    animation-delay: -0.8s;
  }

  @-webkit-keyframes sk-stretchdelay {
    0%,
    40%,
    100% {
      -webkit-transform: scaleY(0.4);
    }
    20% {
      -webkit-transform: scaleY(1);
    }
  }

  @keyframes sk-stretchdelay {
    0%,
    40%,
    100% {
      transform: scaleY(0.4);
      -webkit-transform: scaleY(0.4);
    }
    20% {
      transform: scaleY(1);
      -webkit-transform: scaleY(1);
    }
  }
`;

const Loading = () => (
  <StyledLoading>
    <div className="spinner">
      <div className="rect1" />
      <div className="rect2" />
      <div className="rect3" />
      <div className="rect4" />
      <div className="rect5" />
    </div>
  </StyledLoading>
);

export default Loading;
