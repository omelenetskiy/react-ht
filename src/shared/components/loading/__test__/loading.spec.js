import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-enzyme';
import 'jest-styled-components';

import Loading from '../Loading';

describe('Loading', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Loading />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
