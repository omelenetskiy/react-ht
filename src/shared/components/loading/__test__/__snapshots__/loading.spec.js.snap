// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<Loading__StyledLoading>
  <div
    className="spinner"
  >
    <div
      className="rect1"
    />
    <div
      className="rect2"
    />
    <div
      className="rect3"
    />
    <div
      className="rect4"
    />
    <div
      className="rect5"
    />
  </div>
</Loading__StyledLoading>
`;
